web/source/kmwkeyboards.ts

```diff
@@ -0,0 +1,1383 @@
+            var manager = this;
+            window.setTimeout(function(){
+              manager.installKeyboard(loadingStub);
+            },0);
+          }          
+          this.activeStub=this.keyboardStubs[Ln];
+          return;
+        }
+      }
+      (<any>this.keymanweb)._SetTargDir(this.keymanweb._LastActiveElement);  // I2077 - LTR/RTL timing
+    } 
+
+    var Pk=this.activeKeyboard;  // I3319
+    if(Pk !== null)  // I3363 (Build 301)
+      String.kmwEnableSupplementaryPlane(Pk && ((Pk['KS'] && (Pk['KS'] == 1)) || (Pk['KN'] == 'Hieroglyphic'))); // I3319
+    
+    // Initialize the OSK (provided that the base code has been loaded)
+    osk._Load();
+  }
+
+    /**
+   * Install a keyboard script that has been downloaded from a keyboard server
+   * 
+   *  @param  {Object}  kbdStub   keyboard stub to be loaded.
+   *    
+   **/      
+  installKeyboard(kbdStub: KeyboardStub) {
+    var util = this.keymanweb.util;
+    var osk = this.keymanweb.osk;
+
+    var Lscript = <HTMLScriptElement>(<any>util)._CreateElement('SCRIPT');
+    Lscript.charset="UTF-8";        // KMEW-89
+    Lscript.type = 'text/javascript';
+
+    // Preserve any namespaced IDs by use of the script's id tag attribute!
+    if(this.keymanweb.isEmbedded) {
+      Lscript.id = kbdStub['KI'];
+    }
+
+    var kbdFile = kbdStub['KF'];
+    var kbdLang = kbdStub['KL'];
+    var kbdName = kbdStub['KN'];
+
+    var manager = this;
+
+    // Add a handler for cases where the new <script> block fails to load.
+    Lscript.addEventListener('error', function() {
+      if(kbdStub.asyncLoader.timer !== null) {
+        // Clear the timeout timer.
+        window.clearTimeout(kbdStub.asyncLoader.timer);
+        kbdStub.asyncLoader.timer = null;
+      }
+
+      // We already know the load has failed... why wait?
+      kbdStub.asyncLoader.callback('Cannot find the ' + kbdName + ' keyboard for ' + kbdLang + '.', 'warn');
+      kbdStub.asyncLoader = null;
+    }, false);
+
+    
+    // The load event will activate a newly-loaded keyboard if successful and report an error if it is not.
+    Lscript.addEventListener('load', function() {
+      if(kbdStub.asyncLoader.timer !== null) {
+        // Clear the timeout timer.
+        window.clearTimeout(kbdStub.asyncLoader.timer);
+        kbdStub.asyncLoader.timer = null;
+      }
+
+      // To determine if the load was successful, we'll need to check the keyboard array for our desired keyboard.
+      // Test if keyboard already loaded
+      var kbd = manager.getKeyboardByID(kbdStub['KI']), Li;
+      if(kbd) {  // Is cleared upon a successful load.
+        
+        //Activate keyboard, if it's still the active stub.
+        if(kbdStub == manager.activeStub) {
+          manager.doBeforeKeyboardChange(kbd['KI'],kbdStub['KLC']);
+          manager.activeKeyboard=kbd;
+
+          if((<any>manager.keymanweb)._LastActiveElement != null) {
```

Comment:
Is this temporary casting? I sure hope so ;-)  (please comment TODO if so).

Reply:
It's temporary, as `_LastActiveElement` isn't actually declared on the object.  These casts serve as a temp workaround during the overall refactor process.  I try to keep them relatively minimal, though they're quite helpful for intermediate testing.